tests/components/verisure/test_config_flow.py

```diff
@@ -0,0 +1,357 @@
+        type(mock_verisure.return_value).installations = PropertyMock(
+            return_value=installations
+        )
+        mock_verisure.login.return_value = True
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: "verisure_my_pages@example.com",
+                CONF_GIID: giid,
+                CONF_LOCK_CODE_DIGITS: 10,
+                CONF_LOCK_DEFAULT_CODE: "123456",
+                CONF_PASSWORD: "SuperS3cr3t!",
+            },
+        )
+
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "ascending (12345th street)"
+    assert result["data"] == {
+        CONF_EMAIL: "verisure_my_pages@example.com",
+        CONF_GIID: "12345",
+        CONF_LOCK_CODE_DIGITS: 10,
+        CONF_LOCK_DEFAULT_CODE: "123456",
+        CONF_PASSWORD: "SuperS3cr3t!",
+    }
+
+    assert len(mock_verisure.mock_calls) == 2
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_imports_invalid_login(hass: HomeAssistant) -> None:
+    """Test a YAML import that results in a invalid login."""
+    with patch(
+        "homeassistant.components.verisure.config_flow.Verisure.login",
+        side_effect=VerisureLoginError,
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: "verisure_my_pages@example.com",
+                CONF_GIID: None,
+                CONF_LOCK_CODE_DIGITS: None,
+                CONF_LOCK_DEFAULT_CODE: None,
+                CONF_PASSWORD: "SuperS3cr3t!",
+            },
+        )
+
+    assert result["step_id"] == "user"
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_imports_needs_user_installation_choice(hass: HomeAssistant) -> None:
+    """Test a YAML import that needs to use to decide on the installation."""
+    with patch(
+        "homeassistant.components.verisure.config_flow.Verisure",
+    ) as mock_verisure:
+        type(mock_verisure.return_value).installations = PropertyMock(
+            return_value=TEST_INSTALLATIONS
+        )
+        mock_verisure.login.return_value = True
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_EMAIL: "verisure_my_pages@example.com",
+                CONF_GIID: None,
+                CONF_LOCK_CODE_DIGITS: None,
+                CONF_LOCK_DEFAULT_CODE: None,
+                CONF_PASSWORD: "SuperS3cr3t!",
+            },
+        )
+
+    assert result["step_id"] == "installation"
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] is None
```

Comment:
It's safest to continue the flow in the test to either create entry, error or abort.

Reply:
There is no reason to continue the flow for the tests themselves. I've added a bit that cancels the flow to clean them up.

Reply:
There could be a side effect in this entry point (import) that changes the outcome in the rest of the flow that we don't test now.

Reply:
Alright completed flows in all cases.